Roles/AddOns/Common/Rainbow.cs

```diff
@@ -0,0 +1,59 @@
+using System.Linq;
+using static TOHE.Options;
+
+namespace TOHE.Roles.AddOns.Common
+{
+    public static class Rainbow
+    {
+        private static readonly int Id = 27400;
+        public static OptionItem CrewCanBeRainbow;
+        public static OptionItem ImpCanBeRainbow;
+        public static OptionItem NeutralCanBeRainbow;
+        public static OptionItem RainbowColorChangeCoolDown;
+        public static long LastColorChange;
+        public static void SetupCustomOptions()
+        {
+            SetupAdtRoleOptions(Id, CustomRoles.Rainbow, canSetNum: true, tab: TabGroup.Addons);
+            CrewCanBeRainbow = BooleanOptionItem.Create(Id + 10, "CrewCanBeRainbow", true, TabGroup.Addons, false)
+                .SetParent(CustomRoleSpawnChances[CustomRoles.Rainbow]);
+            ImpCanBeRainbow = BooleanOptionItem.Create(Id + 11, "ImpCanBeRainbow", true, TabGroup.Addons, false)
+                .SetParent(CustomRoleSpawnChances[CustomRoles.Rainbow]);
+            NeutralCanBeRainbow = BooleanOptionItem.Create(Id + 12, "NeutralCanBeRainbow", true, TabGroup.Addons, false)
+                .SetParent(CustomRoleSpawnChances[CustomRoles.Rainbow]);
+            RainbowColorChangeCoolDown = IntegerOptionItem.Create(Id + 13, "RainbowColorChangeCoolDown", new(1, 100, 1), 3, TabGroup.Addons, false)
+                .SetParent(CustomRoleSpawnChances[CustomRoles.Rainbow]);
+        }
+        public static void Init()
+        {
+            LastColorChange = Utils.GetTimeStamp();
+        }
+        public static void OnFixedUpdate()
+        {
+            if (!CustomRoles.Rainbow.RoleExist()) return;
+
```

Comment:
It is better not to use RoleExist() in OnFixedUpdate because it may cause lags, it is better to create an IsEnable (bool variable) and use it